src/workerd/io/internal-subrequest-type.h

```diff
@@ -0,0 +1,7 @@
+#pragma once
+
+#include <kj/one-of.h>
+
+struct GenericInternalSubrequest{};
+struct DOSubrequest{};
+typedef kj::OneOf<GenericInternalSubrequest, DOSubrequest> InternalSubrequestType;
```

Comment:
This could be a flat enum as it stands, but in the future KV might use this system too for their free tier limit enforcement, and they might want to further subdivide their operations. It's possible to collapse those operations into a flat enum nonetheless, but I thought that that would be a bit messy in the future, so I went for a kj::OneOf.

Can change this to an enum if that's deemed to more tasteful.